src/main/offline/io/stargate/sgv2/jsonapi/api/request/FileWriterParams.java

```diff
@@ -0,0 +1,30 @@
+package io.stargate.sgv2.jsonapi.api.request;
+
+import java.util.List;
+
+public record FileWriterParams(
+    String keyspaceName,
+    String tableName,
+    String ssTableOutputDirectory,
+    int fileWriterBufferSizeInMB,
+    String createTableCQL,
+    String insertStatementCQL,
+    List<String> indexCQLs,
+    Boolean vectorEnabled) {
+  public FileWriterParams {
+    if (keyspaceName == null
+        || keyspaceName.isBlank()
+        || tableName == null
+        || tableName.isBlank()
+        || ssTableOutputDirectory == null
+        || ssTableOutputDirectory.isBlank()
+        || fileWriterBufferSizeInMB <= 0
+        || createTableCQL == null
+        || createTableCQL.isBlank()
+        || insertStatementCQL == null
+        || insertStatementCQL.isBlank()
+        || vectorEnabled == null) {
+      throw new IllegalArgumentException("Invalid FileWriterParams");
```

Comment:
Ok for now but probably need to change in near future to indicate exactly which argument was bad.